### DISCUSSION TITLE
Estimate number of UniProtKB entries that will have an associated GO-CAM model?

### DISCUSSION THREAD

**Discussion by dlrice** [geneontology/go-fastapi#109] [Q&A] [answered]
I'm trying to estimate the number of UniProtKB entries that will have an associated GO-CAM model and came up with [this gist](https://gist.github.com/dlrice/3b821c257cce45a2578537cace691e06) to arrive at ~2K. I see there are some non-uniprotkb accessions in the list but I'm just trying to ballpark it. Look reasonable to you or am I missing something?

**Comment by kltm** [comment, marked as answer]
This is likely a good guesstimate, but I'd note that the use of gocam-gps.json is definitely going to be stable as it is strictly internal (hidden) and changeable resource.

Another way to estimate (at least for now) would be grabbing the internally intended noctua_uniprotkb.gpad.gz at https://snapshot.geneontology.org/products/upstream_and_raw_data/ and running through it real fast.

As a quick one-off (I did not check):
```
grep -v '^!' noctua_uniprotkb.gpad | cut -f 12 | grep -E -oh "gomodel\:[[:alnum:]]*_?[[:alnum:]]*_?[[:alnum:]]*" | sort | uniq -c | wc -l
1242
```

**Reply by dlrice** [reply]
Many thanks!
